utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/schemaregistry/azure/SchemaNameTopicMatchStrategy.scala

```diff
@@ -1,60 +1,96 @@
 
-import com.google.common.base.CaseFormat
+import pl.touk.nussknacker.engine.schemedkafka.schemaregistry.azure.SchemaNameTopicMatchStrategy.{
+  FullSchemaNameDecomposed,
+  toSchemaNamingConvention
+}
 
 // TODO: It probable should be configurable: e.g would be nice to have possibility to define static topic -> schemaName map in config
-//       Thanks to that would be possible to use existing schemas that not follow our convention in Nussknacker.
+//       Thanks to that it would be possible to use existing schemas that not follow our convention in Nussknacker.
 //       Also in case of ambiguity (>1 schemas with only different namespaces), we could pick the correct one schema.
+class SchemaNameTopicMatchStrategy(referenceTopicList: List[String]) {
+
+  /**
+    * List all reference topics matching schema names.
+    */
+  def matchAllTopics(fullSchemaNames: List[String], isKey: Boolean): List[String] = {
+    val coreSchemaNames = fullSchemaNames.collect { case _ @FullSchemaNameDecomposed(coreName, _, `isKey`) =>
+      coreName
+    }
+    referenceTopicList.collect {
+      case topicName if coreSchemaNames.contains(toSchemaNamingConvention(topicName)) => topicName
+    }
+  }
+
+  /**
+    * List all schemas matching reference topics.
+    */
+  def matchAllSchemas(fullSchemaNames: List[String], isKey: Boolean): List[String] = {
+    fullSchemaNames.collect {
+      case fullSchemaName @ FullSchemaNameDecomposed(coreName, _, `isKey`)
+          if referenceTopicList.map(toSchemaNamingConvention).contains(coreName) =>
+        fullSchemaName
+    }
+  }
+
+}
+
 object SchemaNameTopicMatchStrategy {
 
   val KeySuffix   = "Key"
   val ValueSuffix = "Value"
-  def topicNameFromKeySchemaName(schemaName: String): Option[String] =
-    FullSchemaNameDecomposed.unapply(schemaName).filter(_._3).map(_._1)
 
-  def topicNameFromValueSchemaName(schemaName: String): Option[String] =
-    FullSchemaNameDecomposed.unapply(schemaName).filterNot(_._3).map(_._1)
+  def apply(referenceTopicList: List[String] = Nil): SchemaNameTopicMatchStrategy = new SchemaNameTopicMatchStrategy(
+    referenceTopicList
+  )
 
   def keySchemaNameFromTopicName(topicName: String): String = schemaNameFromTopicName(topicName, isKey = true)
 
   def valueSchemaNameFromTopicName(topicName: String): String = schemaNameFromTopicName(topicName, isKey = false)
 
   def schemaNameFromTopicName(topicName: String, isKey: Boolean): String = {
     val suffix = if (isKey) KeySuffix else ValueSuffix
-    CaseFormat.LOWER_HYPHEN.to(CaseFormat.UPPER_CAMEL, topicName) + suffix
+    toSchemaNamingConvention(topicName) + suffix
   }
 
+  /**
+    * Transforms topic name to schema naming convention.
+    * @see https://nussknacker.io/documentation/docs/integration/KafkaIntegration/#association-between-schema-with-topic
+    */
+  def toSchemaNamingConvention(topicName: String): String =
+    topicName.toLowerCase.replaceAll("\\W", " ").split(" ").map(_.capitalize).mkString("")
```

Comment:
1. It probably won't change anything but for sake of clarity, IMO better would be to match `\\W+`.
2. I thought about having a `sanitize(topicName)` method that just remove all `\\W+` and do `toLowerCase`. Then similar method `sanitize(coreSchemaName)` that would also do `toLowerCase`. Thanks to that we would match also schema full names like `com.namespace.idontcareaboutcapitalization`. WDYT?

Reply:
Ad 1. fixed

Reply:
Ad 2. let it be as it is